packages/odyssey-react-mui/src/labs/Drawer.tsx

```diff
@@ -0,0 +1,224 @@
+  /**
+   * The title of the Drawer
+   */
+  title?: string;
+  /**
+   * Type of Drawer
+   */
+  variant?: (typeof variantValues)[number];
+  ariaLabel: string;
+} & AllowedProps;
+
+const DrawerHeader = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>`
+  position: sticky;
+  display: flex;
+  top: 0;
+  background-color: ${({ odysseyDesignTokens }) =>
+    odysseyDesignTokens.HueNeutralWhite};
+  justify-content: space-between;
+  align-items: center;
+  margin: 0;
+  padding: ${({ odysseyDesignTokens }) => odysseyDesignTokens.Spacing4} 0;
+  font-family: ${({ odysseyDesignTokens }) =>
+    odysseyDesignTokens.TypographyFamilyHeading};
+  color: ${({ odysseyDesignTokens }) => odysseyDesignTokens.HueNeutral900};
+
+  h2 {
+    margin: 0;
+    font-size: ${({ odysseyDesignTokens }) =>
+      odysseyDesignTokens.TypographySizeHeading5};
+    font-weight: 500;
+  }
+`;
+
+const DrawerFooter = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{
+  odysseyDesignTokens: DesignTokens;
+}>`
+  display: flex;
+  position: sticky;
+  bottom: 0;
+  background-color: ${({ odysseyDesignTokens }) =>
+    odysseyDesignTokens.HueNeutralWhite};
+  flex-direction: row;
+  flex-wrap: nowrap;
+  justify-content: flex-end;
+  align-items: center;
+  padding: ${({ odysseyDesignTokens }) => odysseyDesignTokens.Spacing4};
+  align-content: center;
+`;
+
+const Drawer = ({
+  callToActionFirstComponent,
+  callToActionSecondComponent,
+  callToActionLastComponent,
+  children,
+  isOpen,
+  onClose,
+  testId,
+  title,
+  translate,
+  variant = "temporary",
+  ariaLabel,
+}: DrawerProps) => {
+  const [isContentScrollable, setIsContentScrollable] = useState(false);
+  const drawerontentRef = useRef<HTMLDivElement>(null);
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  //If RTL is set in the theme, align the drawer on the left side of the screen, uses right by default.
+  const i18nRTL = useTranslation();
+  const anchorDirection = i18nRTL.i18n.dir() === "rtl" ? "left" : "right";
+
+  useEffect(() => {
+    let frameId: number;
+
+    const handleContentScroll = () => {
+      const drawerontentElement = drawerontentRef.current;
+      if (drawerontentElement) {
+        setIsContentScrollable(
+          drawerontentElement.scrollHeight > drawerontentElement.clientHeight
+        );
+      }
+      frameId = requestAnimationFrame(handleContentScroll);
+    };
+
+    if (isOpen) {
+      frameId = requestAnimationFrame(handleContentScroll);
+    }
+
+    return () => {
+      cancelAnimationFrame(frameId);
+    };
+  }, [isOpen]);
+
+  const content =
+    typeof children === "string" ? (
+      <Box translate={translate}>{children}</Box>
+    ) : (
+      children
+    );
+
+  return (
+    <MuiDrawer
+      data-se={testId}
+      anchor={anchorDirection}
+      open={isOpen}
+      onClose={onClose}
+      variant={variant}
+      sx={{
```

Comment:
This removes the "slide" transition that MUI applies to Drawer by default for the `persistent` variant as an inline style. We have to do this to use the custom transition for `persistent` defined in [components.tsx](https://github.com/okta/odyssey/pull/2126/files#diff-ff64766022cabbb3dfec5d24e4530df06e384d834b6f5324a2f0f10b0af5cd79)

I don't love using `sx` here but I am not aware of another way to disable the transition, other than adding an override with `!important` in components.tsx like:

`          ...(ownerState.variant === "persistent" && {
            transition: "none !important",
          }),`

Reply:
@chrispulsinelli-okta we could do this in `components.tx` instead but it would require an `!important` 🤮  because MUI adds the `transition` as an inline style when the `Drawer` is rendered. As far as I can determine, there's no programmatic way to stop this so I was weighing which option is the lesser evil.